src/test/py/test_main.py

```diff
@@ -139,8 +149,18 @@ async def test_processed_received_hl7_messages(mocker, caplog):
     mocker.patch.object(mock_hl7_message, "__str__", return_value=hl7_text)
     NATSMessager.send_msg.side_effect = Exception("force exception from mock")
     await main.process_received_hl7_messages(asyncmock_reader, asyncmock_writer)
+
     assert "ack_code='AE'" in str(mock_hl7_message.mock_calls[0])
-    assert "HL7 Listener received a message of type: ADT^A01" in caplog.text
+
```

Comment:
This is cool. I would encourage the team to embrace this type of approach. We can just check "one thing" to ensure that the "structure" of the structure logging is valid. We don't want to do this for everything since that's nonsensical and would be very brittle.